habitat-lab/habitat/core/embodied_task.py

```diff
@@ -295,45 +295,43 @@ def reset(self, episode: Episode):
 
     def _step_single_action(
         self,
-        observations: Any,
         action_name: Any,
         action: Dict[str, Any],
         episode: Episode,
-        is_last_action=True,
     ):
         if isinstance(action_name, (int, np.integer)):
             action_name = self.get_action_name(action_name)
         assert (
             action_name in self.actions
         ), f"Can't find '{action_name}' action in {self.actions.keys()}."
         task_action = self.actions[action_name]
-        observations.update(
-            task_action.step(
-                **action["action_args"],
-                task=self,
-                is_last_action=is_last_action,
-            )
+        return task_action.step(
```

Comment:
Most of the time, task_action will return None, but for some legacy reason, it will sometimes return the observations. This is because sim.step(action) returns observations. In order to avoid rendering observations twice, we will use the observations computed in task_action when they are available.